### DISCUSSION TITLE
How to get the total stake delegated to a pool using the QueryLedgerStateLiveStakeDistribution API?

### DISCUSSION THREAD

**Discussion by pedromtcosta** [CardanoSolutions/ogmios#393] [Q & A] [answered]
Using the pool `pool1qqqqqdk4zhsjuxxd8jyvwncf5eucfskz0xjjj64fdmlgj735lr9` as a random example. When using the `liveStakeDistribution` function from the `LedgerStateQueryClient` I get back the following for this pool:
```
"pool1qqqqqdk4zhsjuxxd8jyvwncf5eucfskz0xjjj64fdmlgj735lr9": {
    "stake": "5214703195371/2842537686040880",
    "vrf": "6de7605b2afd7e205318fc8b4329d26439b5b179b4746fea12f1ce50395f5d9e"
  }
```

When checking this pool on CardanoScan, it shows it has a total stake of about 67M ADA: [https://cardanoscan.io/pool/00000036d515e12e18cd3c88c74f09a67984c2c279a5296aa96efe89](https://cardanoscan.io/pool/00000036d515e12e18cd3c88c74f09a67984c2c279a5296aa96efe89)

Is it possible to get to this number using the Ogmios APIs?

Another thing, on the documentation for this endpoint it says the response is the stake relative to the total stake of the network, but the number on the right keeps changing for each pool:
```
"pool1qqqqqdk4zhsjuxxd8jyvwncf5eucfskz0xjjj64fdmlgj735lr9": {
    "stake": "5214703195371/2842537686040880",
    "vrf": "6de7605b2afd7e205318fc8b4329d26439b5b179b4746fea12f1ce50395f5d9e"
  },
  "pool1qqqqpanw9zc0rzh0yp247nzf2s35uvnsm7aaesfl2nnejaev0uc": {
    "stake": "61711789621259/36952989918531440",
    "vrf": "bb862fcc397de6c06dea9a9ad8442a3d17826e01799a79b861fc083341eaed3d"
  }
```
Based on the docs I was thinking the number on the left would be the total stake on the pool and on the right would be the total stake on the network, but apparently this is not the case. Could you please explain what are each of these numbers?

Thank you!

**Comment by crypto-titan** [comment]
+1

**Comment by KtorZ** [comment, marked as answer]
Try the `rewardsProvenance` query.

> Could you please explain what are each of these numbers

These numbers are a numerator and a denominator. The ratio represents the exact relative stake of the pool in the system. If you want the absolute amount, you need to know the circulating supply of Ada. 

I am actually thinking of removing this query, or more exactly, merge it with the rewardsProvenance one which is more useful and also provides other handy pieces of information.

**Comment by KtorZ** [comment]
One may wonder: but why have such a query to begin with? 

And to answer that, you have to understand that Ogmios is originally a proxy that provides access to the underlying queries available on the node itself. And until v6, I've been trying to stay faithful to the queries provided by the node/ledger. But truth is that many of them do not make for a great and comprehensible API at all. So from v6.0.0 onwards, I've started altering some of those, sometimes merging two queries into one, etc.. in an attempt to make them more useful. 

It only recently occurred to me that this particular one was... not so useful. So I'll soon be deprecating it.

**Comment by pedromtcosta** [comment]
Thank you for the response and clarification @KtorZ, makes perfect sense to me.

I am trying the `rewardsProvenance` query, but it never returns. it has been running for 5 minutes already, not sure if this is expected. The other queries I mentioned are working fine

**Comment by pedromtcosta** [comment]
It actually just finished running, so took about 25 minutes

**Comment by KtorZ** [comment]
Is your node fully synchronized ?

That query does take a bit of time given the number of stake pools, but 25 min sounds a bit excessive.
